.circleci/configs/e2e_test.yml

```diff
@@ -0,0 +1,395 @@
+        type: string
+      flex_ui_version:
+        type: string
+      shell:
+        type: string
+      node-version:
+        type: string
+    environment:
+      NODE_OPTIONS: --max-old-space-size=14336
+      PACKAGE_VERSION: << parameters.package_version >>
+      FLEX_UI_VERSION: << parameters.flex_ui_version >>
+      NODE_VERSION: << parameters.node-version >>
+    executor:
+      name: win/default
+      shell: << parameters.shell >>
+      size: xlarge
+    steps:
+      - checkout
+      - cmd_windows-nvm-install-node:
+          node-version: << parameters.node-version >>
+      - cmd_install-dependencies
+      - cmd_run-build
+      - cmd_run-e2e-tests-win
+      - cmd_run-e2e-tests-win-ts
+      - store_artifacts:
+          path: packages/flex-plugin-e2e-tests/screenshots
+
+  job_e2e-test-macos-node18:
+    resource_class: medium
+    parameters:
+      flex_ui_version:
+        type: string
+      package_version:
+        type: string
+      node-version:
+        type: string
+    environment:
+      NODE_OPTIONS: --max-old-space-size=14336
+      PACKAGE_VERSION: << parameters.package_version >>
+      FLEX_UI_VERSION: << parameters.flex_ui_version >>
+      NODE_VERSION: << parameters.node-version >>
+    executor:
+      name: macos
+    steps:
+      - checkout
+      - node/install:
+          node-version: << parameters.node-version >>
+      - cmd_install-dependencies
+      - cmd_run-build
+      - cmd_set-mac-permissions
+      - cmd_run-e2e-tests-mac-js
+      - cmd_run-e2e-tests-mac-ts
+      - store_artifacts:
+          path: packages/flex-plugin-e2e-tests/screenshots
+
+workflows:
+  workflow_e2e-tester:
+    when: << pipeline.parameters.run_e2e_test >>
+    jobs:
+      - job_e2e-test-linux-node16:
+          matrix:
+            parameters:
+              package_version: [<< pipeline.parameters.package_version >>]
+              flex_ui_version: [<< pipeline.parameters.flex_ui_version >>]
+              node-version: ["16.20.0"]
+      - job_e2e-test-win-node16:
+          matrix:
+            parameters:
+              package_version: [<< pipeline.parameters.package_version >>]
+              flex_ui_version: [<< pipeline.parameters.flex_ui_version >>]
+              shell: ["bash.exe"]
+              node-version: ["16.20.0"]
+      - job_e2e-test-macos-node16:
+          matrix:
+            parameters:
+              package_version: [<< pipeline.parameters.package_version >>]
+              flex_ui_version: [<< pipeline.parameters.flex_ui_version >>]
+              node-version: ["16.20.0"]
+  workflow_nightly-e2e:
+    triggers:
+      - schedule:
+          cron: "0 16 * * *"
+          filters:
+            branches:
+              only:
+                - main
+    jobs:
+      - job_e2e-test-linux-node16:
+          matrix:
+            parameters:
+              package_version: ["latest"]
+              flex_ui_version: ["latest"]
+              node-version: ["16.20.0"]
```

Comment:
Is matrix needed since we have one job per os per node version?

Reply:
Matrix  is not only used for node version but also for package and flex_ui_version. I have kept the matrix  in case we need to use it to run the job with multiple ui_version or package versions in the future.